### DISCUSSION TITLE
How to listen for job completion?

### DISCUSSION THREAD

**Discussion by YeomansIII** [breejs/bree#208] [Q&A] [answered]
I am trying to listen for job completion and perform necessary work on the main service as a result. The jobs will not necessarily return any data using postMessage or be required to include `postMessage('done')`, so that should not be relied upon. The logger already spits out a message when a job goes online and when a job exits and I can use a custom logger to receive this message, however it is in string format and the name of the job would have to be parsed out of the string, which is obviously not ideal.

Is there any other way to listen for the exiting of a worker thread or completion of ANY job, regardless of if it uses `postMessage` or not?

Also, just as a sanity check, when I do use `parentPort.postMessage('done')`, I receive that message, but then the logger never receives the message that the job/thread has exited. Has the job exited? Or is the postMessage('done') method of ending a job broken?

Thanks.

**Comment by shadowgate15** [comment, marked as answer]
[`worker deleted` event](https://github.com/breejs/bree#listening-for-events) indicates that a job has ended.

**Reply by YeomansIII** [reply]
Perfect, thanks. I should have seen this.
